src/psykal/zwx2_kern.f90

```diff
@@ -4,23 +4,24 @@ module zwx2_kern_mod
 !
 contains
   !
-  subroutine zwx2_kern(zwx,zdt,zbtr,pwn,mydomain,zind,zslpx,jk,jj,ji)
+  subroutine zwx2_kern(zwx,pwn,mydomain,zind,zslpx,jk,jj,ji)
     !
     real*8, intent(out) :: zwx(:,:,:)
     real*8, intent(in) :: pwn(:,:,:), mydomain(:,:,:), zind(:,:,:), zslpx(:,:,:)
-    real*8, intent(in) :: zdt,zbtr
-    integer, intent(in) :: jk,jj,ji
+    integer, intent(in) :: jj,ji,jk
     ! local variables
-    real*8 :: z0w, zalpha, zw, zzwx, zzwy
+    real*8 :: z0w, zalpha, zw, zzwx, zzwy, zdt, zbtr
     !
-    z0w = SIGN( 0.5d0, pwn(ji,jj,jk+1) )
+    zdt=1
+    zbtr=1.
+    z0w = SIGN( 0.5d0, pwn(ji,jj,jk) )
     zalpha = 0.5d0 + z0w
-    zw  = z0w - 0.5d0 * pwn(ji,jj,jk+1) * zdt * zbtr
+    zw  = z0w - 0.5d0 * pwn(ji,jj,jk) * zdt * zbtr
     
-    zzwx = mydomain(ji,jj,jk+1) + zind(ji,jj,jk) * (zw * zslpx(ji,jj,jk+1))
-    zzwy = mydomain(ji,jj,jk  ) + zind(ji,jj,jk) * (zw * zslpx(ji,jj,jk  ))
+    zzwx = mydomain(ji,jj,jk  ) + zind(ji,jj,jk-1) * (zw * zslpx(ji,jj,jk  ))
+    zzwy = mydomain(ji,jj,jk-1) + zind(ji,jj,jk-1) * (zw * zslpx(ji,jj,jk-1))
     
-    zwx(ji,jj,jk+1) = pwn(ji,jj,jk+1) * ( zalpha * zzwx + (1.-zalpha) * zzwy )
+    zwx(ji,jj,jk) = pwn(ji,jj,jk) * ( zalpha * zzwx + (1.-zalpha) * zzwy )
```

Comment:
Indexing changes so kernel writes to `(ji,jj,jk)` rather than `(ji,jj,jk+1)` - good.